src/RouteMap.ts

```diff
@@ -59,6 +61,58 @@ export default class RouteMap {
         }
     }
 
+    /**
+     * Get the name of the method which can handle this route but also
+     * overrides event.pathParameters based on data extracted from the url
+     */
+    public getRouteOverridePathParams(event: APIGatewayProxyEvent): string | undefined{
+        for (const [controllerPathKey, controllerKey] of this.map.entries()) {
+            // isPathMatch overrides the current event.pathParams
+            if (this.isPathMatch(controllerPathKey, event)) {
+                return controllerKey;
+            }
+        }
+    }
+
+    /**
+     * Checks if the event.path matches one the url patterns avaible on
+     * the controller map.
+     * TODO: allow snake cased pattern match
```

Comment:
@oyanezm Did you intend to leave this TODO in? (I don't mind, but just checking)

Reply:
@johnmikk in the meantime yes :(, left the message for the future if someone (me) can take it